File: app/src/main/java/com/zjgsu/ai/calibrationtest/Calibration.java
package com.zjgsu.ai.calibrationtest;

import android.graphics.RectF;

import org.json.JSONException;
import org.json.JSONObject;

/**
 * Created by Double on 2017/4/24.
 */

public class Calibration {

    private static final String JSON_CATEGORY = "category";
    private static final String JSON_AREA = "area";
    private static final String JSON_PHOTO = "path";

    private static int count = 0;

    private String category;
    private Area mArea;
    private Photo src;

    public String getCategory() {
        return category;
    }

    public Calibration(String category, String path) {
        this.category = category;
        this.src = new Photo(path);
    }

    public Calibration(JSONObject json) throws JSONException {
        category = json.getString(JSON_CATEGORY);
        src = new Photo(json.getJSONObject(JSON_PHOTO));
        if (json.has(JSON_AREA))
            mArea = new Area(json.getJSONObject(JSON_AREA));
    }

    public JSONObject toJSON() throws JSONException {
        JSONObject json = new JSONObject();
        json.put(JSON_CATEGORY, category);
        json.put(JSON_PHOTO, src.toJSON());
        if (mArea != null)
            json.put(JSON_AREA, mArea.toJSON());
        return json;
    }

    public void setCategory(String category) {
        this.category = category;
    }

    public void setSrc(Photo src) {
        this.src = src;
    }

    public void setArea(Area area) {
        this.mArea = area;
    }

    public Area getArea() {
        return mArea;
    }

    public RectF[] getAreaRects() {
        if (mArea != null) {
            return mArea.getRects();
        } else
            return null;
    }

    public String getPhotoPath() {
        return src.getPath();
    }

    @Override
    public String toString() {
        return category;
    }
}

File: app/src/main/java/com/zjgsu/ai/calibrationtest/Calibration.kt
package com.zjgsu.ai.calibrationtest

import org.json.JSONObject

/**
 * Created by Double on 30/07/2017.
 */

class Calibration private constructor() {

    companion object {
        private val JSON_CATEGORY: String = "category"
        private val JSON_AREA: String = "area"
        private val JSON_PHOTO: String = "path"

        private var count: Int = 0
    }

    private lateinit var category: String
    fun getCategory() = category
    fun setCategory(c: String) { category = c }

    private lateinit var mArea: Area
    fun getArea() = mArea
    fun setArea(area: Area) { mArea = area }

    private var src: Photo? = null
    fun setSrc(p: String) { src = Photo(p) }

    fun getPhotoPath() = src?.getPath()

    fun getAreaRects() = mArea?.getRects()

    constructor(category: String, path: String) : this() {
        this.category = category
        this.src = Photo(path)
    }

    constructor(json: JSONObject) : this() {
        category = json.getString(JSON_CATEGORY)
        src = Photo(json.getJSONObject(JSON_PHOTO))
        if (json.has(JSON_AREA))
            mArea = Area(json.getJSONObject(JSON_AREA))
    }

    fun toJSON(): JSONObject {
        var json: JSONObject = JSONObject()
        json.put(JSON_CATEGORY, category)
        json.put(JSON_PHOTO, src?.toJSON())
        json.put(JSON_AREA, mArea?.toJSON())

        return json
    }

    override fun toString() = category

}